src/resources/rmt_esp32.cc

```diff
@@ -695,5 +1031,45 @@ PRIMITIVE(sync_manager_reset) {
   return process->null_object();
 }
 
+PRIMITIVE(encoder_new) {
+  ARGS(SimpleResourceGroup, group, Blob, bytes)
+  if (bytes.length() == 0) FAIL(INVALID_ARGUMENT);
+  if (!RmtPatternEncoder::validate(bytes.address(), bytes.length())) FAIL(INVALID_ARGUMENT);
+
+  ByteArray* proxy = process->object_heap()->allocate_proxy();
+  if (proxy == null) FAIL(ALLOCATION_FAILED);
+
+  // The encoder is called from interrupts and must be in SRAM.
+  uint8* buffer = unvoid_cast<uint8*>(heap_caps_malloc(bytes.length(), SRAM_CAPS));
+  if (!buffer) FAIL(MALLOC_FAILED);
+  bool handed_to_encoder = false;
+  Defer del_buffer { [&] { if (!handed_to_encoder) free(buffer); } };
+
+  memcpy(buffer, bytes.address(), bytes.length());
+
+  auto encoder_memory = heap_caps_malloc(sizeof(RmtPatternEncoder), SRAM_CAPS);
+  if (!encoder_memory) FAIL(MALLOC_FAILED);
+  auto encoder = new (encoder_memory) RmtPatternEncoder(buffer);
```

Comment:
Nit: It annoys me a little bit that this is allocated with a ref count of zero - given that it is automatically deleted when the ref count reaches zero. Not sure if there is a major elegant way to express that the encoder is always owned by a resource when it is first created.

Reply:
done.